pkg/envoy/embedded_envoy.go

```diff
@@ -104,7 +104,6 @@ func startEmbeddedEnvoy(config embeddedEnvoyConfig) (*EmbeddedEnvoy, error) {
 	bootstrapFilePath := filepath.Join(config.runDir, "envoy", "bootstrap.pb")
 
 	writeBootstrapConfigFile(bootstrapConfig{
-		// Use the same structure as Istio's pilot-agent for the node ID: nodeType~ipAddress~proxyId~domain
```

Comment:
I would retain a comment about the `nodeId` format (inherited from Istio), or then simplify it to something like "Cilium-host-proxy" etc.

Reply:
i added back a small comment about having this inherited from Istio. I agree that we could simplify the name. Especially if we decide to delete the need for "extracting the "IP" from the Envoy Node ID.

But i'd like to have this in a separate PR to keep the context smaller (the other would also impact the DaemonSet mode where we set the node id in the Helm YAMLs.